pkg/csi/service/vanilla/controller_helper.go

```diff
@@ -231,3 +231,55 @@ func getVolumeManagerForVolumeID(ctx context.Context, controller *controller, vo
 	}
 	return volumeManager, nil
 }
+
+// getVCenterManagerForVolumeID returns vCenter manager for the given volumeId.
+// If multi-vcenter-csi-topology feature is disabled, legacy vCenter manager is returned
+// from `controller.manager.VCenterManager`.
+// If multi-vcenter-csi-topology feature is enabled, vCenter manager is returned from
+// `controller.managers.VCenterManager``.
+func getVCenterManagerForVolumeID(ctx context.Context, controller *controller, volumeId string,
+	volumeInfoService cnsvolumeinfo.VolumeInfoService) (vsphere.VirtualCenterManager, error) {
+	var vCenterManager vsphere.VirtualCenterManager
+	if multivCenterCSITopologyEnabled {
+		vCenterManager = controller.managers.VcenterManager
+	} else {
+		vCenterManager = controller.manager.VcenterManager
+	}
+	return vCenterManager, nil
+}
+
+// getVCenterHostForVolumeID returns VC host for the given volumeId.
+// If multi-vcenter-csi-topology feature is disabled legacy VC host is returned
+// from `controller.manager.VCenterConfig.Host`.
+// If multi-vcenter-csi-topology feature is enabled but volumeInfoService is not instantiated VC host
+// is returned from controller.managers.VCenterConfig.Host for `controller.managers.CnsConfig.Global.VCenterIP`.
+// If multi-vcenter-csi-topology feature is enabled and volumeInfoService is instantiated VC host is returned for
+// the vCenter IP mapped to the requested VolumeID from volumeInfoService.
+func getVCenterHostForVolumeID(ctx context.Context, controller *controller, volumeId string,
+	volumeInfoService cnsvolumeinfo.VolumeInfoService) (string, error) {
+	var vCenterConfig *vsphere.VirtualCenterConfig
+	var vCenterHost string
+	var err error
+	var vcHostFound bool
+	log := logger.GetLogger(ctx)
+	if multivCenterCSITopologyEnabled {
+		if len(controller.managers.VcenterConfigs) > 1 {
+			// Multi vCenter Deployment
+			vCenterHost, err = volumeInfoService.GetvCenterForVolumeID(ctx, volumeId)
+			if err != nil {
+				return "", logger.LogNewErrorCodef(log, codes.Internal,
+					"failed to get vCenter for the volumeID: %q with err=%+v", volumeId, err)
+			}
+		} else {
+			// Single vCenter Deployment
+			if vCenterConfig, vcHostFound = controller.managers.VcenterConfigs[controller.managers.CnsConfig.Global.VCenterIP]; !vcHostFound {
+				return "", logger.LogNewErrorCodef(log, codes.Internal,
+					"could not get vCenterConfig for vCenter: %q", controller.managers.CnsConfig.Global.VCenterIP)
+			}
+			vCenterHost = vCenterConfig.Host
```

Comment:
I think vCenterConfig.Host is going to be same as `controller.managers.CnsConfig.Global.VCenterIP` as there is only 1 VC.

Reply:
when multi-vc is enabled let's use `managers.VcenterConfigs`.
@skogta we should not use `c.manager.VcenterConfig.Host` here.